File: braggbnb108_ui/src/services/BookingRequestService.js
import http from "../http-common"; 

class BookingRequestService {
  getAllBookingRequests(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/bookingRequest/bookingRequests`, searchDTO);
  }

  get(bookingRequestId) {
    return this.getRequest(`/bookingRequest/${bookingRequestId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/bookingRequest?field=${matchData}`, null);
  }

  addBookingRequest(data) {
    return http.post("/bookingRequest/addBookingRequest", data);
  }

  update(data) {
  	return http.post("/bookingRequest/updateBookingRequest", data);
  }
  
  uploadImage(data,bookingRequestId) {
  	return http.postForm("/bookingRequest/uploadImage/"+bookingRequestId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new BookingRequestService();

File: src/main/java/com/braggbnb108/service/BookingRequestService.java
package com.braggbnb108.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb108.domain.BookingRequest;
import com.braggbnb108.dto.BookingRequestDTO;
import com.braggbnb108.dto.BookingRequestSearchDTO;
import com.braggbnb108.dto.BookingRequestPageDTO;
import com.braggbnb108.dto.BookingRequestConvertCriteriaDTO;
import com.braggbnb108.service.GenericService;
import com.braggbnb108.dto.common.RequestDTO;
import com.braggbnb108.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface BookingRequestService extends GenericService<BookingRequest, Integer> {

	List<BookingRequest> findAll();

	ResultDTO addBookingRequest(BookingRequestDTO bookingRequestDTO, RequestDTO requestDTO);

	ResultDTO updateBookingRequest(BookingRequestDTO bookingRequestDTO, RequestDTO requestDTO);

    Page<BookingRequest> getAllBookingRequests(Pageable pageable);

    Page<BookingRequest> getAllBookingRequests(Specification<BookingRequest> spec, Pageable pageable);

	ResponseEntity<BookingRequestPageDTO> getBookingRequests(BookingRequestSearchDTO bookingRequestSearchDTO);
	
	List<BookingRequestDTO> convertBookingRequestsToBookingRequestDTOs(List<BookingRequest> bookingRequests, BookingRequestConvertCriteriaDTO convertCriteria);

	BookingRequestDTO getBookingRequestDTOById(Integer bookingRequestId);







}





